### PR TITLE
Fixed ICore GetSupportedProperties

### PR DESCRIPTION
**Pull request by apankratovantonp**
### Details:
 - *item1*
 - *...*

### Tickets:
 - *79092*

### CODE REVIEW

File: docs/template_plugin/tests/functional/shared_tests_instances/behavior/ov_executable_network/properties.cpp [hunk 1]
```diff
@@ -0,0 +1,120 @@
...
+                ::testing::ValuesIn(properties)),
+        OVCompiledModelPropertiesTests::getTestCaseName);
+
+INSTANTIATE_TEST_SUITE_P(smoke_Hetero_BehaviorTests, OVCompiledModelPropertiesTests,
+        ::testing::Combine(
+                ::testing::Values(CommonTestUtils::DEVICE_HETERO),
+                ::testing::ValuesIn(hetero_properties)),
+        OVCompiledModelPropertiesTests::getTestCaseName);
+
+INSTANTIATE_TEST_SUITE_P(DISABLED_smoke_Multi_BehaviorTests, OVCompiledModelPropertiesTests,
```
Comment (ilya-lavrenov) [inline docs/template_plugin/tests/functional/shared_tests_instances/behavior/ov_executable_network/properties.cpp:109]: why these tests are disabled?

File: docs/template_plugin/tests/functional/shared_tests_instances/behavior/ov_executable_network/properties.cpp [hunk 2]
```diff
@@ -0,0 +1,120 @@
...
+
+const std::vector<ov::AnyMap> multi_properties = {
+        {ov::device::priorities(CommonTestUtils::DEVICE_TEMPLATE), ov::enable_profiling(true)},
+        {ov::device::priorities(CommonTestUtils::DEVICE_TEMPLATE), ov::device::id("0")},
+};
+
+const std::vector<ov::AnyMap> auto_batch_properties = {
+        {{CONFIG_KEY(AUTO_BATCH_DEVICE_CONFIG) , CommonTestUtils::DEVICE_TEMPLATE}},
+        {{CONFIG_KEY(AUTO_BATCH_DEVICE_CONFIG) , CommonTestUtils::DEVICE_TEMPLATE},
+         {CONFIG_KEY(AUTO_BATCH_TIMEOUT) , "1"}},
```
Comment (ilya-lavrenov) [inline docs/template_plugin/tests/functional/shared_tests_instances/behavior/ov_executable_network/properties.cpp:94]: why new properties are not used here?

File: docs/template_plugin/tests/functional/shared_tests_instances/behavior/ov_executable_network/properties.cpp [hunk 3]
```diff
@@ -0,0 +1,120 @@
...
+
+const std::vector<ov::AnyMap> hetero_properties = {
+        {ov::device::priorities(CommonTestUtils::DEVICE_TEMPLATE), ov::enable_profiling(true)},
+        {ov::device::priorities(CommonTestUtils::DEVICE_TEMPLATE), ov::device::id("0")},
+};
+
+
+const std::vector<ov::AnyMap> multi_properties = {
+        {ov::device::priorities(CommonTestUtils::DEVICE_TEMPLATE), ov::enable_profiling(true)},
+        {ov::device::priorities(CommonTestUtils::DEVICE_TEMPLATE), ov::device::id("0")},
```
Comment (ilya-lavrenov) [inline docs/template_plugin/tests/functional/shared_tests_instances/behavior/ov_executable_network/properties.cpp:88]: is "0" as device ID supported? I see that for TEMPLATE we defined it as:
```c++
        std::vector<std::string> availableDevices = {""};
```

File: src/inference/src/ie_core.cpp [hunk 4]
```diff
@@ -1113,7 +1143,18 @@ class CoreImpl : public ie::ICore, public std::enable_shared_from_this<ie::ICore
 
     std::map<std::string, std::string> GetSupportedConfig(const std::string& deviceName,
                                                           const std::map<std::string, std::string>& configs) override {
-        std::vector<std::string> supportedConfigKeys = GetMetric(deviceName, METRIC_KEY(SUPPORTED_CONFIG_KEYS));
+        std::vector<std::string> supportedConfigKeys;
+        try {
+            auto supported_properties = ICore::get_property(deviceName, ov::supported_properties);
```
Comment (ilya-lavrenov) [inline src/inference/src/ie_core.cpp:1148]: `ov::supported_properties` does not provide old properties for most of the plugins. So, we need to combine mutable properties from `ov::supported_properties` plus `METRIC_KEY(SUPPORTED_CONFIG_KEYS)`

File: docs/template_plugin/tests/functional/shared_tests_instances/behavior/ov_executable_network/properties.cpp (hunk 2 above)
Comment (apankratovantonp) [reply docs/template_plugin/tests/functional/shared_tests_instances/behavior/ov_executable_network/properties.cpp:94]: @myshevts do auto batch provide new properties for this configuration values?

File: docs/template_plugin/tests/functional/shared_tests_instances/behavior/ov_executable_network/properties.cpp (hunk 3 above)
Comment (apankratovantonp) [reply docs/template_plugin/tests/functional/shared_tests_instances/behavior/ov_executable_network/properties.cpp:88]: template plugin has default separate `device_id` property that dose not affect anything.

**Review by ilya-lavrenov — APPROVED** [review]

**Review by myshevts — APPROVED** [review]

File: src/inference/src/ie_core.cpp [hunk 5]
```diff
@@ -1113,7 +1143,13 @@ class CoreImpl : public ie::ICore, public std::enable_shared_from_this<ie::ICore
 
     std::map<std::string, std::string> GetSupportedConfig(const std::string& deviceName,
                                                           const std::map<std::string, std::string>& configs) override {
-        std::vector<std::string> supportedConfigKeys = GetMetric(deviceName, METRIC_KEY(SUPPORTED_CONFIG_KEYS));
+        auto supportedConfigKeys =
+            GetMetric(deviceName, METRIC_KEY(SUPPORTED_CONFIG_KEYS)).as<std::vector<std::string>>();
+        for (auto&& property : ICore::get_property(deviceName, ov::supported_properties)) {
```
Comment (ilya-lavrenov) [inline src/inference/src/ie_core.cpp:1148]: if this method is called from old API, we need to return only old keys.

File: docs/template_plugin/tests/functional/shared_tests_instances/behavior/ov_executable_network/properties.cpp (hunk 2 above)
Comment (ilya-lavrenov) [reply docs/template_plugin/tests/functional/shared_tests_instances/behavior/ov_executable_network/properties.cpp:94]: per @myshevts 
let's add explicit test that AUTO:CPU supports old keys (for e.g. streams) via new API.

Comment (myshevts) [reply docs/template_plugin/tests/functional/shared_tests_instances/behavior/ov_executable_network/properties.cpp:94]: > per @myshevts let's add explicit test that AUTO:CPU supports old keys (for e.g. streams) via new API.
new config key thru new API, but for the AUTO:CPU

**Pull request merged**